docs/keto/modeling/learn-the-concepts.mdx

```diff
@@ -0,0 +1,360 @@
+    Folder -->|editor| User
+    Folder -->|viewer| User
+`}
+/>
+
+Putting this all together in one graph:
+
+<Mermaid
+  chart={`
+graph LR
+    Document -->|owner| User
+    Document -->|editor| User
+    Document -->|viewer| User
+    Document -->|parent| Folder
+    Folder -->|parent| Folder
+    Folder -->|owner| User
+    Folder -->|editor| User
+    Folder -->|viewer| User
+`}
+/>
+
+### Define relations in the OPL
+
+Next, add each relation to the permission config.
+
+:::tip
+
+The relations from the description translate to relationships in Ory Permissions, which are declared inside the corresponding
+class in the Ory Permission Language. Since you store arbitrary tuples in Ory Permissions, you always have many-to-many relations
+between these objects. To reflect this in our language, the convention is to pluralize the relation name, i.e., `viewers` instead
+of `viewer`.
+
+:::
+
+```ts title="permissions-v2.ts"
+import { Namespace, Context } from "@ory/keto-namespace-types"
+
+class User implements Namespace {}
+
+class Document implements Namespace {
+  // highlight-start
+  related: {
+    owners: User[]
+    editors: User[]
+    viewers: User[]
+    parents: Folder[]
+  }
+  // highlight-end
+}
+
+class Folder implements Namespace {
+  // highlight-start
+  related: {
+    owners: User[]
+    editors: User[]
+    viewers: User[]
+    parents: Folder[]
+  }
+  // highlight-end
+}
+```
+
+### List permissions to check for each object
+
+When you perform an action on behalf of a user, you should check a specific permission, such as `view` or `edit`, instead of a
+relationship, such as `owner`. Therefore, each permission is derived from relationships and other permissions.
+
+Add these permissions to the model to express what the application needs. For our document storage, you have these permissions:
+
+- `view` a document if the user is a `viewer`, `editor`, or `owner` of the document; or if the user can `view` the parent folder
+- `edit` a document if the user is a `editor`, or `owner` of the document; or if the user can `edit` the parent folder
+- `delete` a document if the user is an `owner` of the document; or if the user can `delete` the parent folder
+- `share` a document if the user is an `owner` of the document; or if the user can `share` the parent folder
+- `delete` a folder if the user is an `owner` of the folder; or if the user can `delete` the parent folder
+- `share` a folder if the user is an `owner` of the folder; or if the user can `share` the parent folder
+
+### Define permissions in the OPL
+
+The permissions are expressed in the OPL as TypeScript functions that take a context containing the subject, for example the
+current user, and that answer the permission check based on the relationships this object has, in this case the current document.
+
+:::tip
+
+The permissions from the description are declared as functions inside the `permits` property of the corresponding class in the Ory
+Permission Language.
```

Comment:
You introduced the OPL contraction, it's okay to use it instead of the full name.